### DISCUSSION TITLE
`time` module

### DISCUSSION THREAD

**Discussion by benstigsen** [zuri-lang/zuri#156] [Ideas]
It would be useful if there was a `time` module for Blade.
I'd like to implement this if possible, some things that it'd be useful for:
```js
var start = time.now()
time.sleep(5 * time.MINUTE)
var elapsed = time.now() - start

echo "Time elapsed ${elapsed}"
```

I do realize that there's `date`, `time()`, `microtime()` and `os.sleep()`, but a simple `time` module would be very nice, and also help people coming from other languages.

Perhaps doing the following:
```
os.sleep()  -> time.sleep()
time()      -> time.time()
microtime() -> time.microtime()
```

**Comment by mcfriend99** [comment]
Hmm... You are suggesting to move the time functions into the same module. This is a welcomed idea. Please feel free to submit a PR for this. It's highly welcomed.

Take note that in your PR, it will be good to update the `tests`, `modules` and `benchmarks` that use `time()`, `microtime()` and `os.sleep()` currently. And the documentation will best align with the existing documentation format to make it easy to generate documentations for the website.

**Comment by benstigsen** [comment]
Awesome! I'll start working on it, together with `tests`, `modules` and so on. Then we can take a look at it together when I make a PR _soon_.

**Comment by mcfriend99** [comment]
There is a Nyssa package for testing called `qi`. It's far from perfect, but it's working.
